### PR TITLE
Symbol api

### PR DESCRIPTION
**Pull request by mortenvp**

### CODE REVIEW

**Comment by mortenvp** [comment]
@JuanCabre seems everything is green but for a bit of formatting, perhaps you can try to fix it.. you can see the violating lines here:
http://buildbot.steinwurf.com/builders/kodo-python-astyle/builds/996/steps/fab%20astyle.check_code/logs/stdio

**Comment by JuanCabre** [comment]
Hello Morten,

Before I work on the formatting, there is something wrong, and I cannot understand the problem. Perhaps you can give me some insights.

I wanted to test the write_symbol method. I create an encoder with 4 symbols and symbol_size of 1.
I create a payload with it. Then, I copy the coefficients from that payload and call the write_symbol method with those coefficients. The output of that method doesn't match the output of the write_payload method. 

This is a minimal code to reproduce it:

```python
import os
import kodo

symbols = 4
symbol_size = 1
field = kodo.field.binary8

encoder_factory = kodo.RLNCEncoderFactory(field, symbols, symbol_size)
encoder = encoder_factory.build()

data_in = bytearray(os.urandom(encoder.block_size()))
encoder.set_const_symbols(data_in)
encoder.set_systematic_off()

payload = encoder.write_payload()

# These are the coding coefficients used to generate the payload. It runs from 1
# because the element 0 is the systematic on/off flag
coeff = payload[1:encoder.symbols()+1].copy()

print("payload", payload)
print("Coefficients", coeff)

symbol = encoder.write_symbol(coeff)
print("Symbol", symbol)

if symbol != payload[encoder.symbols()+1:]:
    print("There is a problem")
```

**Comment by petya2164** [comment]
@JuanCabre @mortenvp Hello guys, I just wanted to ask what is the status of this branch?
I am currently replacing boost-python with pybind11 here: https://github.com/steinwurf/kodo-python/pull/81 (obviously, this will conflict with any changes in .cpp files)
If the Symbol API is almost finished, then we should probably merge both to avoid future conflicts.

@JuanCabre I see a fundamental problem with your example above:
You are assuming that the RLNC encoder uses the full_vector format to store the coefficients in the payload. By default, it uses the seed format, so you will only see the random seed in the payload (no coefficients).
In any case, we will need a good Python example before we can merge this. We should implement something like this: https://github.com/steinwurf/kodo-rlnc/blob/master/examples/encode_decode_using_coefficients/encode_decode_using_coefficients.cpp#L81-L91

File: src/kodo_python/decoder.hpp [hunk 1]
```diff
@@ -53,6 +53,19 @@ PyObject* decoder_write_payload(Decoder& decoder)
     return PyByteArray_FromStringAndSize((char*)payload.data(), length);
 }
 
+template<class Decoder>
+PyObject* decoder_write_symbol(Decoder& decoder, PyObject* coefficients)
```
Comment (petya2164) [inline src/kodo_python/decoder.hpp]: This function should be removed. The decoder does not provide a write_symbol method. 
See the decoder's Symbol API here: https://github.com/steinwurf/kodo-rlnc/blob/master/src/kodo_rlnc/decoder.hpp#L177-L191

**Pull request merged**